fecgov/fec-cms
# 4905 Remove Prettier

## Summary

- Resolves #4905 

As part of the Webpack upgrade, we're switching fully to ESLint, relying on it to do its current linting work, but also to do the formatting work that Prettier has been doing.

### Required reviewers

2 front-end types, feel free to swap

## Impacted areas of the application

The overall build and automated tests including
- build
- build-production
- format
- lint, and
- test-single

Updated the rules in .eslintrc to better match our current rules from Prettier

Added another .eslintrc file to the tests directory to get rid of the errors when VSC would complain that it didn't know what 'describe', 'it', etc. were.

Also included instructions to update VSC to follow these rules and a link to instructions for Sublime.

## Screenshots

n/a

## Related PRs

none

## How to test

- pull the branch like normal
- `pip install -r requirements.txt` if you'd like. (I had to install psycopg2-binary to silence errors
- `npm i` because we're updating several packages
- `npm run build` (no errors)
- `npm run build-production` (no errors)
- `npm run format` (0 errors, 3 warnings about ignored files)
- `npm run lint` (no errors)
- `npm run test-single` (no errors. Usual warnings about 'VERSION', EventEmitters, and Browserify)
- Open any JavaScript file (except in */vendor*/*)
- Make a syntax error of some kind (not just a formatting issue, but something like an early return, unending loop, undeclared vars…). Save the file.
   - If you're using VSC, it should underline it as soon as you break it
   - Run `npm run lint` and make sure your new error is pointed out to you
- Make a formatting error of some kind (wrong number of spaces, incorrect single/double quotes, etc). Save the file.
   - If you're using VSC, it should underline the error as you're making it
   - Hovering over the error should tell you what's wrong and show you and icon.
   - Clicking that icon should give you a chance to fix that error or all of them like it. (If it's an error it can decipher)
   - Save the file.
   - Run `npm run format` and make sure your broken file is listed along with where and why.
   - Run `npm run format --fix`. Did it fix your error for you?

## Update IDEs

### Sublime

- Full directions at https://packagecontrol.io/packages/ESLint
- (Use shift-cmd-p, not shift-cntrl-p)
- I couldn't get Sublime to check files as I typed or as I saved files but I'm pretty sure it's possible. Whether you want that is another matter
- This is my ESLint.sublime-settings file:
```
{
  "tab_size": 2,
  "node_path": "[whatever `which node` tells you]",
}
```


### Visual Studio Code
- Go to Extensions
- Disable anything that mentions Prettier
- Install ESLint (by Dirk Baeumer)
- Some of my extension settings for ESLint:
   - Show Documentation: enabled
   - Mode: 'all'
   - Enable: checked
   - On Ignored Files: 'off'
   - Package Manager: 'npm'
- If there are any questions about making ESLint the default, yes, do.
